packages/integration/src/github/GithubCredentialsProvider.ts

```diff
@@ -121,22 +120,15 @@ class GithubAppManager {
     });
   }
 
+  getInstallations(): Promise<
+    RestEndpointMethodTypes['apps']['listInstallations']['response']['data']
+  > {
+    return this.appClient.paginate(this.appClient.apps.listInstallations);
+  }
+
   private async getInstallationData(owner: string): Promise<InstallationData> {
-    // List all installations using the last used etag.
-    // Return cached InstallationData if error with status 304 is thrown.
-    try {
-      this.installations = await this.appClient.apps.listInstallations({
-        headers: {
-          'If-None-Match': this.installations?.headers.etag,
-          Accept: HEADERS.Accept,
-        },
-      });
-    } catch (error) {
-      if (error.status !== 304) {
-        throw error;
-      }
-    }
-    const installation = this.installations?.data.find(
+    const allInstallations = await this.getInstallations();
```

Comment:
I realize this isn't part of your pr, but it feels to me that maybe this querying should be cached and only done at most once per X time unit. If I read things correctly, we reach out to the github api over and over again, only to grab a single one out of the list, and then the caching logic comes after that.

Well, something for future improvement perhaps